### PR TITLE
adding implicit check to prevent segfault

### PR DESCRIPTION
**Pull request by WallyMaier**
## Proposed Changes
Adding an implicit flag to prevent segfault while running Roe scheme explicitly.
 
## Related Work
N/A

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x ] I am submitting my contribution to the develop branch.
- [x ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [ x] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.

### CODE REVIEW

File: SU2_CFD/src/numerics_direct_mean.cpp [hunk 1]
```diff
@@ -3274,9 +3274,11 @@ void CUpwRoe_Flow::ComputeResidual(su2double *val_residual, su2double **val_Jaco
   if (RoeSoundSpeed2 <= 0.0) {
     for (iVar = 0; iVar < nVar; iVar++) {
       val_residual[iVar] = 0.0;
-      for (jVar = 0; jVar < nVar; jVar++) {
-        val_Jacobian_i[iVar][iVar] = 0.0;
-        val_Jacobian_j[iVar][iVar] = 0.0;
+      if (implicit){
+        for (jVar = 0; jVar < nVar; jVar++) {
+          val_Jacobian_i[iVar][iVar] = 0.0;
```
Comment (pcarruscag) [inline SU2_CFD/src/numerics_direct_mean.cpp]: Why do we loop through jVar but access only the diagonal here? This is either a bug or an unnecessary loop.

Comment (vdweide) [reply SU2_CFD/src/numerics_direct_mean.cpp]: It is a bug, also present in the original code. One of the iVar should be jVar. 

The question is whether this treatment of negative square of the speed of sound (setting the flux to zero) is the best one possible (if there already is a correct treatment). Maybe a dissipation term based on a^* would be better in this case (if it is only used during the iteration process).

Comment (pcarruscag) [reply SU2_CFD/src/numerics_direct_mean.cpp]: Agreed. Would clamping RoeSoundSpeed2 to some safe lower bound be a solution? Setting Jacobian contributions to 0 may make it ill-conditioned.

Comment (vdweide) [reply SU2_CFD/src/numerics_direct_mean.cpp]: At least better than what is currently in there, although these tricks usually only postpone the crash.

**Pull request merged**